docs/runbooks/runbook-examples/terraform/provision-aws-with-terraform.md

```diff
@@ -0,0 +1,526 @@
+---
+title: Provision AWS resources with Terraform
+description: With runbooks, you can use Terraform to create resources in AWS.
+position: 20
+---
+
+AWS CloudFormation is a great tool to use to provisions resources, however, it doesn't keep track of state.  With runbooks, you can use Terraform to provision resources on AWS as well as keep them in the desired state.
+
+The following example will use Terraform to dynamically create worker machines based on auto-scaling rules.  Instead of defining the Terraform template directly in the step template, this example will make use of a package.  The package will consist of the following files:
+- autoscaling.tf
+- autoscalingpolicy.tf
+- backend.tf
+- installTentacle.sh
+- provider.tf
+- securitygroup.tf
+- vars.tf
+- vpc.tf
+
+The different AWS resources types have been separated into their respective files to make it easier to maintain.
+
+## autoscaling.tf
+
+This file contains the definitions for creating the auto-scaling configuration in AWS:
+
+<details>
```

Comment:
This is a cool idea. I changed the text from 'file contents' to the name of the file. Unless our styles prevent it, it should be clear you can expand the section. I'll merge and test.